### PR TITLE
Llama2 FT

### PR DESCRIPTION
**Pull request by poonehmousavi**
## What does this PR do?

Finetuning Lamma2 for response generation task using MultiWOZ dataset.
List any dependencies that are required for this change.
 - accelerate
 - bitsandbytes
 - peft
- sacrebleu

### CODE REVIEW

**Comment by poonehmousavi** [comment]
I have added two versions: llama2.py is in the old format and llama2_new.py is in the new version. Also, I added a new version of huggingface_interface_new.py. which I added the quantization config and AUtomodelCasualLM. Let me know what you think of the changes @mhn226 .. When we finalize the best for this interface, I will update the recipe accordingly

File: speechbrain/lobes/models/huggingface_transformers/huggingface_new.py [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+        model: AutoModel
+            HuggingFace generic model class.
+        save_path : str
+            Path (dir) of the downloaded model.
+        cache_dir : str
+            Path (dir) in which a downloaded pretrained model configuration should be cached.
+        """
+        is_sb, ckpt_file, is_local = self._check_model_source(source, save_path)
+        if is_sb:
+            config = config.from_pretrained(source, cache_dir=save_path,quantization_config=quantization_config)
```
Comment (mhn226) [inline speechbrain/lobes/models/huggingface_transformers/huggingface_new.py:163]: @poonehmousavi : Do we also need quantization_config in the case when is_sb = False, i,e, adding quantization_config in the below else condition?

**Comment by mravanelli** [comment]
Thank you @poonehmousavi for this PR. It's a valuable addition to SpeechBrain. Here are my comments from a first review of the code:

1. **Failing Tests:**
   - It appears there are some failing tests, mainly related to style issues. 

2. **Permission Issue in Recipe Tests:**
   - The recipe tests seem to be failing due to a permission issue. 
      ```python
      python -c 'from tests.utils.recipe_tests import run_recipe_tests; print("TEST FAILED!") if not(run_recipe_tests(filters_fields=["Dataset"], filters=[["MultiWOZ"]], do_checks=True, run_opts="--device=cuda")) else print("TEST PASSED")'
      ```
I got the same issue when running the llama2 recipe. Are you able to replicate it?

3. **README Correction:**
   - In the README file, there's a small typo in the directory path. Please correct:
      ```bash
      cd recipes/MultiWOZ/response_generation[LLM_model] => cd recipes/MultiWOZ/response_generation/[LLM_model]
      ```
@mhn226  do you have additional comments?

**Comment by mhn226** [comment]
> Thank you @poonehmousavi for this PR. It's a valuable addition to SpeechBrain. Here are my comments from a first review of the code:
> 
>     1. **Failing Tests:**
>        
>        * It appears there are some failing tests, mainly related to style issues.
> 
>     2. **Permission Issue in Recipe Tests:**
>        
>        * The recipe tests seem to be failing due to a permission issue.
>          ```python
>          python -c 'from tests.utils.recipe_tests import run_recipe_tests; print("TEST FAILED!") if not(run_recipe_tests(filters_fields=["Dataset"], filters=[["MultiWOZ"]], do_checks=True, run_opts="--device=cuda")) else print("TEST PASSED")'
>          ```
> 
> 
> I got the same issue when running the llama2 recipe. Are you able to replicate it?
> 
>     3. **README Correction:**
>        
>        * In the README file, there's a small typo in the directory path. Please correct:
>          ```shell
>          cd recipes/MultiWOZ/response_generation[LLM_model] => cd recipes/MultiWOZ/response_generation/[LLM_model]
>          ```
> 
> 
> @mhn226 do you have additional comments?

Hi @mravanelli , maybe come back here next week because currently we're trying to figure out what is the best option for doing this. The PR is rather a draft version for now ^^
And yes for llama2 we need to ask meta for the access. So for testing we might need to skip or maybe use [this](https://huggingface.co/daryl149/llama-2-7b-chat-hf).

**Comment by poonehmousavi** [comment]
I closed this PR and transfer all the changes to this  [PR](https://github.com/speechbrain/speechbrain/pull/2299)

**Pull request closed without merging**